ui/revenuecatui/src/main/kotlin/com/revenuecat/purchases/ui/revenuecatui/data/processed/VariableProcessor.kt

```diff
@@ -86,6 +69,34 @@ internal object VariableProcessor {
         }
     }
 
+    private fun processVariable(
+        variableName: VariableName,
+        variableDataProvider: VariableDataProvider,
+        rcPackage: Package,
+        locale: Locale,
+    ) = when (variableName) {
+        VariableName.APP_NAME -> variableDataProvider.applicationName
+        VariableName.PRICE -> variableDataProvider.localizedPrice(rcPackage)
+        VariableName.PRICE_PER_PERIOD -> variableDataProvider.localizedPricePerPeriod(rcPackage, locale)
+        VariableName.TOTAL_PRICE_AND_PER_MONTH -> variableDataProvider.localizedPriceAndPerMonth(
+            rcPackage,
+            locale,
+        )
+
+        VariableName.PRODUCT_NAME -> variableDataProvider.productName(rcPackage)
+        VariableName.SUB_PERIOD -> variableDataProvider.periodName(rcPackage)
+        VariableName.SUB_PRICE_PER_MONTH -> variableDataProvider.localizedPricePerMonth(
+            rcPackage,
+            locale,
+        )
+
+        VariableName.SUB_DURATION -> variableDataProvider.subscriptionDuration(rcPackage, locale)
+        VariableName.SUB_OFFER_DURATION -> variableDataProvider.introductoryOfferDuration(rcPackage, locale)
+        VariableName.SUB_OFFER_DURATION_2 -> variableDataProvider.introductoryOfferDuration(rcPackage, locale)
```

Comment:
Should this pick only the intro phase and not the free phase and then the intro phase? Currently both this and sub_offer_duration will return the same value right?

Reply:
Oh right, if `SUB_OFFER_DURATION` picks `option?.introPhase`, `SUB_OFFER_DURATION_2` will also pick `option?.introPhase`. Is that what you mean?

Reply:
oh wait, nevermind, this is the duration 😮‍💨

Reply:
Well yeah, in that case we should probably pick the pricing phases directly and get them in order. The first one would likely be the free trial, the second one the intro price

Reply:
I realized `option.introPhase` and `option.freePhase` also check if the amounts are 0L, so I think it's better to use those